### DISCUSSION TITLE
error build vslam on ROS

### DISCUSSION THREAD

**Discussion by adamardiyanto** [stella-cv/stella_vslam#381] [Q&A] [answered]
hello i facing error while i building run_slam.cc for vslam_ros. i use ubuntu 18.04 and ros melodic. has anyone facing this error too? 

/home/adamardi/stella_ws/src/stella_vslam_ros/src/run_slam.cc: In function ‘void tracking(const std::shared_ptr<stella_vslam::config>&, const string&, const string&, bool, const string&, bool)’:
/home/adamardi/stella_ws/src/stella_vslam_ros/src/run_slam.cc:32:14: error: ‘using element_type = class stella_vslam::config {aka class stella_vslam::config}’ has no member named ‘camera_’
     if (cfg->camera_->setup_type_ == stella_vslam::camera::setup_type_t::Monocular) {
              ^~~~~~~
/home/adamardi/stella_ws/src/stella_vslam_ros/src/run_slam.cc:35:19: error: ‘using element_type = class stella_vslam::config {aka class stella_vslam::config}’ has no member named ‘camera_’
     else if (cfg->camera_->setup_type_ == stella_vslam::camera::setup_type_t::Stereo) {
                   ^~~~~~~
/home/adamardi/stella_ws/src/stella_vslam_ros/src/run_slam.cc:38:19: error: ‘using element_type = class stella_vslam::config {aka class stella_vslam::config}’ has no member named ‘camera_’
     else if (cfg->camera_->setup_type_ == stella_vslam::camera::setup_type_t::RGBD) {
                   ^~~~~~~
/home/adamardi/stella_ws/src/stella_vslam_ros/src/run_slam.cc:42:64: error: ‘using element_type = class stella_vslam::config {aka class stella_vslam::config}’ has no member named ‘camera_’
         throw std::runtime_error("Invalid setup type: " + cfg->camera_->get_setup_type_string());
                                                                ^~~~~~~
src/CMakeFiles/run_slam.dir/build.make:62: recipe for target 'src/CMakeFiles/run_slam.dir/run_slam.cc.o' failed
make[2]: *** [src/CMakeFiles/run_slam.dir/run_slam.cc.o] Error 1
CMakeFiles/Makefile2:3170: recipe for target 'src/CMakeFiles/run_slam.dir/all' failed
make[1]: *** [src/CMakeFiles/run_slam.dir/all] Error 2
Makefile:140: recipe for target 'all' failed
make: *** [all] Error 2

**Comment by ymd-stella** [comment, marked as answer]
Update is in progress.

**Reply by ymd-stella** [reply]
Done.

**Reply by adamardiyanto** [reply]
it seem fixed but i am facing another problem here while building this package again.

[ 33%] Building CXX object src/CMakeFiles/run_slam.dir/run_slam.cc.o
/home/adamardi/stella_ws/src/stella_vslam_ros/src/run_slam.cc:20:10: fatal error: ghc/filesystem.hpp: No such file or directory
 #include <ghc/filesystem.hpp>
          ^~~~~~~~~~~~~~~~~~~~
compilation terminated.
src/CMakeFiles/run_slam.dir/build.make:62: recipe for target 'src/CMakeFiles/run_slam.dir/run_slam.cc.o' failed
make[2]: *** [src/CMakeFiles/run_slam.dir/run_slam.cc.o] Error 1
CMakeFiles/Makefile2:3168: recipe for target 'src/CMakeFiles/run_slam.dir/all' failed
make[1]: *** [src/CMakeFiles/run_slam.dir/all] Error 2
Makefile:140: recipe for target 'all' failed
make: *** [all] Error 2

**Reply by ymd-stella** [reply]
Clone stella_vslam_ros with `git clone --recursive stella_vslam_ros` or download submodules with `git submodule update --init --recursive`.

**Reply by mgrallos** [reply]
`Hi i tried the first time but it says that package does not exist 
`

**Comment by adamardiyanto** [comment]
i've done clone this package with that methods, but there is another error when i build this package.

[ 66%] Building CXX object src/CMakeFiles/run_slam.dir/stella_vslam_ros.cc.o
/home/adamardi/stella_ws/src/stella_vslam_ros/src/run_slam.cc: In function ‘void tracking(const std::shared_ptr<stella_vslam_ros::system>&, const std::shared_ptr<stella_vslam::config>&, bool, const string&)’:
/home/adamardi/stella_ws/src/stella_vslam_ros/src/run_slam.cc:40:171: error: no matching function for call to ‘pangolin_viewer::viewer::viewer(YAML::Node, std::shared_ptr<stella_vslam::system>*, const std::shared_ptr<stella_vslam::publish::frame_publisher>, const std::shared_ptr<stella_vslam::publish::map_publisher>)’
     pangolin_viewer::viewer viewer(stella_vslam::util::yaml_optional_ref(cfg->yaml_node_, "PangolinViewer"), &SLAM, SLAM->get_frame_publisher(), SLAM->get_map_publisher());
                                                                                                                                                                           ^
In file included from /home/adamardi/stella_ws/src/stella_vslam_ros/src/run_slam.cc:2:0:
/usr/local/include/pangolin_viewer/viewer.h:37:5: note: candidate: pangolin_viewer::viewer::viewer(const YAML::Node&, stella_vslam::system*, const std::shared_ptr<stella_vslam::publish::frame_publisher>&, const std::shared_ptr<stella_vslam::publish::map_publisher>&)
     viewer(const YAML::Node& yaml_node, stella_vslam::system* system,
     ^~~~~~
/usr/local/include/pangolin_viewer/viewer.h:37:5: note:   no known conversion for argument 2 from ‘std::shared_ptr<stella_vslam::system>*’ to ‘stella_vslam::system*’
src/CMakeFiles/run_slam.dir/build.make:62: recipe for target 'src/CMakeFiles/run_slam.dir/run_slam.cc.o' failed
make[2]: *** [src/CMakeFiles/run_slam.dir/run_slam.cc.o] Error 1
CMakeFiles/Makefile2:3168: recipe for target 'src/CMakeFiles/run_slam.dir/all' failed
make[1]: *** [src/CMakeFiles/run_slam.dir/all] Error 2
Makefile:140: recipe for target 'all' failed
make: *** [all] Error 2
<== Failed to process package 'stella_vslam_ros': 
  Command '['/home/adamardi/stella_ws/devel_isolated/stella_vslam/env.sh', 'make', '-j4', '-l4']' returned non-zero exit status 2

Reproduce this error by running:
==> cd /home/adamardi/stella_ws/build_isolated/stella_vslam_ros && /home/adamardi/stella_ws/devel_isolated/stella_vslam/env.sh make -j4 -l4

Command failed, exiting.

**Reply by ymd-stella** [reply]
Thanks for the report. I will fix it right away.

**Reply by ymd-stella** [reply]
Done.
stella_vslam has also been updated and needs to rebuild.

**Reply by adamardiyanto** [reply]
thank you for your work sir. i ve build it without error. really appreciate it

**Comment by mgrallos** [comment]
Hi @ymd-stella, I hope this finds you well. I was also trying to install stella_vslam but encountered these errors. https://www.loom.com/share/334ac7b619a84f2e90fe2d3d62e79bc7

**Reply by ymd-stella** [reply]
Create another question.
